<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/vektor/hiroken-ros-pkg/hiro_common/urdf/hiro.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="hiro" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="Grey2">
    <color rgba="0.9 0.9 0.9 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0.1 0.1 0.1 1.0"/>
  </material>
  <material name="LightGrey">
    <color rgba="0.6 0.6 0.6 1.0"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="yellow">
    <color rgba="1 1 0 1"/>
  </material>
  <material name="magenta">
    <color rgba="1 0 1 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="cyan">
    <color rgba="0 1 1 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  <material name="blue">
    <color rgba="0 0 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="violet">
    <color rgba="0.6 0 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.6 0 1"/>
  </material>
  <material name="grey">
    <color rgba="0.6 0.6 0.6 1"/>
  </material>
  <material name="spring_green">
    <color rgba="0 1 0.498 1"/>
  </material>
  <material name="chartreuse">
    <color rgba="0.498 1 0 1"/>
  </material>
  <material name="olive_drab">
    <color rgba="0.753 1 0.243 1"/>
  </material>
  <material name="chest_red">
    <texture filename="package://hiro_common/materials/textures/chest_texture.png"/>
  </material>
  <material name="chest_red_2">
    <texture filename="package://hiro_common/materials/textures/chest_texture_2.png"/>
  </material>
  <material name="chest_red_3">
    <texture filename="package://hiro_common/materials/textures/chest_texture_4.jpg"/>
  </material>
  <link name="link_base">
    <inertial>
      <origin xyz="-0.00038809 0.00000166 1.06466017"/>
      <mass value="93.11081000"/>
      <!-- momentsOfInertia [0.03473180 0.00000578 -0.00003341; 0.00000578 0.03474816 0.00000063; -0.00003343 0.00000062 0.00438238] -->
      <inertia ixx="0.03473180" ixy="0.00000578" ixz="-0.00003341" iyy="0.03474816" iyz="0.00000063" izz="0.00438238"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/base.stl"/>
        <!--cylinder length="${base_height+torso_height}" radius="0.030"/-->
      </geometry>
      <origin rpy="0 -1.57079632679 -1.57079632679" xyz="0 0 0.9"/>
      <!--origin xyz="0 0 ${(base_height+torso_height)/2}" rpy="0 0 0"/-->
      <material name="grey"/>
    </visual>
    <collision>
      <geometry>
        <!--mesh filename="package://hiro_common/meshes/base.stl"/-->
        <!--cylinder length="${base_height+torso_height}" radius="0.030"/-->
        <cylinder length="0.3" radius="0.030"/>
      </geometry>
      <!--origin xyz="0 0 ${base_height}" rpy="0 ${-0.5*pi} ${-0.5*pi}"/-->
      <!--origin xyz="0 0 ${(base_height+torso_height)/2}" rpy="0 0 0"/-->
      <origin rpy="0 0 0" xyz="0 0 1.05"/>
      <!--This is safer because this collision model is above the table (not in collision)-->
    </collision>
  </link>
  <gazebo reference="link_base">
    <material>Gazebo/Red</material>
    <mu1 value="100.0"/>
    <mu2 value="100.0"/>
    <kp value="1000000.0"/>
    <kd value="1.0"/>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_chest_yaw" type="revolute">
    <parent link="link_base"/>
    <child link="link_chest_yaw"/>
    <origin rpy="0 0 0" xyz="0 0 0.9"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2.967059725" upper="2.967059725" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-170*rad_per_deg}" soft_upper_limit="${170*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.12e-01 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_chest_yaw">
    <inertial>
      <origin xyz="-0.00493378 -0.00097682 0.34405202"/>
      <mass value="6.20761000"/>
      <!-- momentsOfInertia [0.06395752 -0.00022833 0.00011172; -0.00022833 0.03108451 0.00006480; 0.00011172 0.00006480 0.05011067] -->
      <inertia ixx="0.06395752" ixy="-0.00022833" ixz="0.00011172" iyy="0.03108451" iyz="0.00006480" izz="0.05011067"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/cy.stl"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <!--material name="magenta"/-->
      <material name="chest_red_3"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/cy.stl"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_chest_yaw">
    <material>Gazebo/Red</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_head_yaw" type="revolute">
    <parent link="link_chest_yaw"/>
    <child link="link_neck_yaw"/>
    <origin rpy="0 0 0" xyz="0.0 0 0.5495"/>
    <!-- rotation 0 0 1 0 -->
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2.8797932625" upper="2.8797932625" velocity="0.5"/>
    <safety_controller k_position="0" k_velocity="0" soft_lower_limit="-2.7925268" soft_upper_limit="2.7925268"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_neck_yaw">
    <inertial>
      <origin xyz="-0.00002292 0.00547848 -0.00784016"/>
      <mass value="0.31792900"/>
      <!-- momentsOfInertia [0.00027682 0.00000003 -0.00000033; 0.00000003 0.00013009 -0.00001092; -0.00000033 -0.00001092 0.00020954] -->
      <inertia ixx="0.00027682" ixy="0.00000003" ixz="-0.00000033" iyy="0.00013009" iyz="-0.00001092" izz="0.00020954"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/ny.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/ny.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_neck_yaw">
    <material>Gazebo/Red</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_head_pitch" type="revolute">
    <parent link="link_neck_yaw"/>
    <child link="link_neck_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <!-- rotation 0 1 0 0 -->
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-0.872664625" upper="0.872664625" velocity="0.5"/>
    <safety_controller k_position="0" k_velocity="0" soft_lower_limit="-0.7853981625" soft_upper_limit="0.7853981625"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_neck_pitch">
    <inertial>
      <origin xyz="0.00000502 -0.01667768 0.01631533"/>
      <mass value="0.08085931"/>
      <!-- momentsOfInertia [0.00008070 -0.00000003 -0.00000000; -0.00000003 0.00003025 -0.00001277; -0.00000000 -0.00001277 0.00006637] -->
      <inertia ixx="0.00008070" ixy="-0.00000003" ixz="-0.00000000" iyy="0.00003025" iyz="-0.00001277" izz="0.00006637"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/np.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/np.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_neck_pitch">
    <material>Gazebo/Red</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_rshoulder_yaw" type="revolute">
    <parent link="link_chest_yaw"/>
    <child link="link_rshoulder_yaw"/>
    <origin rpy="0.261799 0 0" xyz="0.000 -0.145 0.380"/>
    <!-- rotation   1 0 0  0.261799 -->
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.570796325" upper="1.570796325" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-85*rad_per_deg}" soft_upper_limit="${85*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.12e-01 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_rshoulder_yaw">
    <inertial>
      <origin xyz="-0.00226388 0.00521383 0.01560807"/>
      <mass value="1.32626000"/>
      <!-- momentsOfInertia [0.00207537 -0.00011276 0.00008927; -0.00011276 0.00240295 0.00004570; 0.00008927 0.00004570 0.00141912] -->
      <inertia ixx="0.00207537" ixy="-0.00011276" ixz="0.00008927" iyy="0.00240295" iyz="4.57e-05" izz="0.00141912"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rsy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rsy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_rshoulder_yaw">
    <material>Gazebo/Red</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_rshoulder_pitch" type="revolute">
    <parent link="link_rshoulder_yaw"/>
    <child link="link_rshoulder_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14159265" upper="2.0943951" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-175*rad_per_deg}" soft_upper_limit="${115*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.68e-01 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_rshoulder_pitch">
    <inertial>
      <origin xyz="-0.00005236 -0.05831338 -0.10706059"/>
      <mass value="1.14898000"/>
      <!-- momentsOfInertia [0.01541678 -0.00000097 0.00000352; -0.00000098 0.01471175 -0.00213705; 0.00000351 -0.00213704 0.00126575] -->
      <inertia ixx="0.01541678" ixy="-9.7e-07" ixz="0.00000352" iyy="0.01471175" iyz="-0.00213705" izz="0.00126575"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rsp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rsp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_rshoulder_pitch">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_relbow_pitch" type="revolute">
    <parent link="link_rshoulder_pitch"/>
    <child link="link_relbow_pitch"/>
    <origin rpy="0 0 0" xyz="0.020 -0.085 -0.250"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-2.7925268" upper="0.174532925" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-155*rad_per_deg}" soft_upper_limit="${0*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	6.00e-02 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_relbow_pitch">
    <inertial>
      <origin xyz="-0.00002254 -0.00167107 -0.07151163"/>
      <mass value="0.57751838"/>
      <!-- momentsOfInertia [0.00143966 0.00000030 0.00000120; 0.00000030 0.00141763 0.00007596; 0.00000120 0.00007596 0.00019720] -->
      <inertia ixx="0.00143966" ixy="3e-07" ixz="0.00000120" iyy="0.00141763" iyz="7.596e-05" izz="0.00019720"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rep.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rep.stl"/>
        <turnGravityOff>true</turnGravityOff>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_relbow_pitch">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <joint name="joint_rwrist_yaw" type="revolute">
    <parent link="link_relbow_pitch"/>
    <child link="link_rwrist_yaw"/>
    <origin rpy="0 0 0" xyz="-0.02 0.0 -0.22"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.4033920375" upper="2.3561944875" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-190*rad_per_deg}" soft_upper_limit="${130*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	3.74e-03 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_rwrist_yaw">
    <inertial>
      <origin xyz="-0.00001788 0.00395812 -0.20273050"/>
      <mass value="0.45741788"/>
      <!-- momentsOfInertia [0.00069205 -0.00000003 0.00000052; -0.00000003 0.00068903 0.00001915; 0.00000052 0.00001915 0.00013509] -->
      <inertia ixx="0.00069205" ixy="-3e-08" ixz="0.00000052" iyy="0.00068903" iyz="1.915e-05" izz="0.00013509"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rwy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rwy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_rwrist_yaw">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="joint_rwrist_pitch" type="revolute">
    <parent link="link_rwrist_yaw"/>
    <child link="link_rwrist_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.74532925" upper="1.74532925" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-95*rad_per_deg}" soft_upper_limit="${95*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.50e-02 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_rwrist_pitch">
    <inertial>
      <origin xyz="0.00243131 -0.00379733 -0.05770359"/>
      <mass value="0.41843443"/>
      <!-- momentsOfInertia [0.00052016 -0.00000169 0.00000790; -0.00000169 0.00050569 0.00005377; 0.00000790 0.00005377 0.00014004] -->
      <inertia ixx="0.00052016" ixy="-1.69e-06" ixz="0.00000790" iyy="0.00050569" iyz="5.377e-05" izz="0.00014004"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rwp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/rwp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_rwrist_pitch">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_rwrist_roll" type="revolute">
    <parent link="link_rwrist_pitch"/>
    <child link="link_rwrist_roll"/>
    <!--origin xyz="0.0  0.0 -0.090" rpy="0 0 0"/-->
    <origin rpy="0 0 0" xyz="-0.025  0.0 -0.090"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-2.268928025" upper="2.268928025" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-125*rad_per_deg}" soft_upper_limit="${125*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	7.56e-03 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_rwrist_roll">
    <inertial>
      <origin xyz="-0.04847597 3.2e-06 -0.03889875"/>
      <mass value="1.05156000"/>
      <!-- momentsOfInertia [0.00194072 -0.00000011 -0.00042482; -0.00000011 0.00209392 -0.00000012; -0.00042482 -0.00000012 0.00035788] -->
      <inertia ixx="0.00194072" ixy="-1.1e-07" ixz="-0.00042482" iyy="0.00209392" iyz="-1.2e-07" izz="0.00035788"/>
    </inertial>
    <visual>
      <geometry>
        <!--mesh filename="package://hiro_common/meshes/wr.stl"/-->
        <box size="0.010 0.055 0.070"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <!--mesh filename="package://hiro_common/meshes/wr.stl"/-->
        <box size="0.010 0.055 0.070"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_rwrist_roll">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="joint_rhand_palm" type="fixed">
    <parent link="link_rwrist_roll"/>
    <child link="link_rhand_palm"/>
    <!--origin xyz="${(-1*(0.130/2))+(-1*(0.010/2))} ${(-1*(0.055/2))+(-1*(0.035/2))} 0" rpy="0 0 0"/-->
    <origin rpy="0 0 0" xyz="-0.07 -0.045 0"/>
    <!-- 
		    Origin x is shifted a half of x-dimension of hand_palm plus a half of x-dimension of link_${side}wrist_roll in the appropriate direction
		    Origin y is shifted a half of y-dimension of hand_palm plus a half of y-dimension of link_${side}wrist_roll in the appropriate direction		    
		  -->
  </joint>
  <link name="link_rhand_palm">
    <visual>
      <geometry>
        <box size="0.130 0.035 0.100"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <material name="magenta"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.130 0.035 0.100"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint_lshoulder_yaw" type="revolute">
    <parent link="link_chest_yaw"/>
    <child link="link_lshoulder_yaw"/>
    <origin rpy="-0.261799 0 0" xyz="0.000 0.145 0.380"/>
    <!-- rotation   1 0 0  0.261799 -->
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.570796325" upper="1.570796325" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-85*rad_per_deg}" soft_upper_limit="${85*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.12e-01 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lshoulder_yaw">
    <inertial>
      <origin xyz="-0.00226388 -0.00521383 0.01560807"/>
      <mass value="1.32626000"/>
      <!-- momentsOfInertia [0.00207537 -0.00011276 0.00008927; -0.00011276 0.00240295 0.00004570; 0.00008927 0.00004570 0.00141912] -->
      <inertia ixx="0.00207537" ixy="0.00011276" ixz="0.00008927" iyy="0.00240295" iyz="-4.57e-05" izz="0.00141912"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lsy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lsy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lshoulder_yaw">
    <material>Gazebo/Red</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_lshoulder_pitch" type="revolute">
    <parent link="link_lshoulder_yaw"/>
    <child link="link_lshoulder_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14159265" upper="2.0943951" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-175*rad_per_deg}" soft_upper_limit="${115*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.68e-01 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lshoulder_pitch">
    <inertial>
      <origin xyz="-0.00005236 0.05831338 -0.10706059"/>
      <mass value="1.14898000"/>
      <!-- momentsOfInertia [0.01541678 -0.00000097 0.00000352; -0.00000098 0.01471175 -0.00213705; 0.00000351 -0.00213704 0.00126575] -->
      <inertia ixx="0.01541678" ixy="9.7e-07" ixz="0.00000352" iyy="0.01471175" iyz="0.00213705" izz="0.00126575"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lsp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lsp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lshoulder_pitch">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_lelbow_pitch" type="revolute">
    <parent link="link_lshoulder_pitch"/>
    <child link="link_lelbow_pitch"/>
    <origin rpy="0 0 0" xyz="0.020 0.085 -0.250"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-2.7925268" upper="0.174532925" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-155*rad_per_deg}" soft_upper_limit="${0*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	6.00e-02 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lelbow_pitch">
    <inertial>
      <origin xyz="-0.00002254 0.00167107 -0.07151163"/>
      <mass value="0.57751838"/>
      <!-- momentsOfInertia [0.00143966 0.00000030 0.00000120; 0.00000030 0.00141763 0.00007596; 0.00000120 0.00007596 0.00019720] -->
      <inertia ixx="0.00143966" ixy="-3e-07" ixz="0.00000120" iyy="0.00141763" iyz="-7.596e-05" izz="0.00019720"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lep.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lep.stl"/>
        <turnGravityOff>true</turnGravityOff>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lelbow_pitch">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <joint name="joint_lwrist_yaw" type="revolute">
    <parent link="link_lelbow_pitch"/>
    <child link="link_lwrist_yaw"/>
    <origin rpy="0 0 0" xyz="-0.02 0.0 -0.22"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.4033920375" upper="2.3561944875" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-190*rad_per_deg}" soft_upper_limit="${130*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	3.74e-03 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lwrist_yaw">
    <inertial>
      <origin xyz="-0.00001788 -0.00395812 -0.20273050"/>
      <mass value="0.45741788"/>
      <!-- momentsOfInertia [0.00069205 -0.00000003 0.00000052; -0.00000003 0.00068903 0.00001915; 0.00000052 0.00001915 0.00013509] -->
      <inertia ixx="0.00069205" ixy="3e-08" ixz="0.00000052" iyy="0.00068903" iyz="-1.915e-05" izz="0.00013509"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lwy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lwy.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lwrist_yaw">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="joint_lwrist_pitch" type="revolute">
    <parent link="link_lwrist_yaw"/>
    <child link="link_lwrist_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.74532925" upper="1.74532925" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-95*rad_per_deg}" soft_upper_limit="${95*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	1.50e-02 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lwrist_pitch">
    <inertial>
      <origin xyz="0.00243131 0.00379733 -0.05770359"/>
      <mass value="0.41843443"/>
      <!-- momentsOfInertia [0.00052016 -0.00000169 0.00000790; -0.00000169 0.00050569 0.00005377; 0.00000790 0.00005377 0.00014004] -->
      <inertia ixx="0.00052016" ixy="1.69e-06" ixz="0.00000790" iyy="0.00050569" iyz="-5.377e-05" izz="0.00014004"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lwp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://hiro_common/meshes/lwp.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lwrist_pitch">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <joint name="joint_lwrist_roll" type="revolute">
    <parent link="link_lwrist_pitch"/>
    <child link="link_lwrist_roll"/>
    <!--origin xyz="0.0  0.0 -0.090" rpy="0 0 0"/-->
    <origin rpy="0 0 0" xyz="-0.025  0.0 -0.090"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-2.268928025" upper="2.268928025" velocity="0.5"/>
    <!--safety_controller  soft_lower_limit="${-125*rad_per_deg}" soft_upper_limit="${125*rad_per_deg}" k_position="0" k_velocity="0" /-->
    <!-- rotorInertia	7.56e-03 -->
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="link_lwrist_roll">
    <inertial>
      <origin xyz="-0.04847597 -3.2e-06 -0.03889875"/>
      <mass value="1.05156000"/>
      <!-- momentsOfInertia [0.00194072 -0.00000011 -0.00042482; -0.00000011 0.00209392 -0.00000012; -0.00042482 -0.00000012 0.00035788] -->
      <inertia ixx="0.00194072" ixy="1.1e-07" ixz="-0.00042482" iyy="0.00209392" iyz="1.2e-07" izz="0.00035788"/>
    </inertial>
    <visual>
      <geometry>
        <!--mesh filename="package://hiro_common/meshes/wr.stl"/-->
        <box size="0.010 0.055 0.070"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <!--mesh filename="package://hiro_common/meshes/wr.stl"/-->
        <box size="0.010 0.055 0.070"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="link_lwrist_roll">
    <material>Gazebo/Yellow</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="joint_lhand_palm" type="fixed">
    <parent link="link_lwrist_roll"/>
    <child link="link_lhand_palm"/>
    <!--origin xyz="${(-1*(0.130/2))+(-1*(0.010/2))} ${(-1*(0.055/2))+(-1*(0.035/2))} 0" rpy="0 0 0"/-->
    <origin rpy="0 0 0" xyz="-0.07 0.045 0"/>
    <!-- 
		    Origin x is shifted a half of x-dimension of hand_palm plus a half of x-dimension of link_${side}wrist_roll in the appropriate direction
		    Origin y is shifted a half of y-dimension of hand_palm plus a half of y-dimension of link_${side}wrist_roll in the appropriate direction		    
		  -->
  </joint>
  <link name="link_lhand_palm">
    <visual>
      <geometry>
        <box size="0.130 0.035 0.100"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <material name="magenta"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.130 0.035 0.100"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
  </link>
</robot>
